### PR TITLE
dockertar-sha256-helper: vendor gzip and flate from golang1.6

### PR DESCRIPTION
**Pull request by giuseppe**
The implementation changed in golang1.8 and the returned checksum is
different.  Vendor the version from golang 1.6 so that we can produce
the same checksum.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by ashcrow** [comment]
@giuseppe I assume the reason to use 1.6 over the latest release is for backwards compatibility?

**Comment by giuseppe** [comment]
if we change the version (and the checksum) then layers that were pulled in the past need to be pulled again.  The gzip/deflate implementation in golang 1.6 gives the same checksum as Docker (did?)

**Comment by ashcrow** [comment]
@giuseppe I see. Makes sense.

**Comment by giuseppe** [comment]
the tests keep failing for a timeout

**Comment by ashcrow** [comment]
bot, test this please

**Comment by ashcrow** [comment]
I've asked @jlebon if there is any known issues with f26 tester.

**Comment by ashcrow** [comment]
bot, retest this please

File: .papr.yml [hunk 1]
```diff
@@ -51,3 +51,5 @@ packages:
   - atomic
   - python3-coverage
   - rpm-build
+  - findutils
+  - make
```
Comment (yuqi-zhang) [inline .papr.yml]: I didn't know that github complained about missing newline

Comment (ashcrow) [inline .papr.yml]: I think it's just noting that there isn't a newline after.

**Comment by ashcrow** [comment]
FWIW it looks like moby doesn't vendor anything in the ``compress`` package.

**Comment by ashcrow** [comment]
bot, retest this please

**Comment by yuqi-zhang** [comment]
Hm, seems like same failure as last time, where its stuck trying to pull the centos image from docker hub. Trying on a f26 atomic host myself I cannot reproduce this, strange...

**Comment by ashcrow** [comment]
@yuqi-zhang there is a known issue right now. The best we can do is keep on top of it and try until it passes.

**Comment by ashcrow** [comment]
@giuseppe I have two commits in https://github.com/ashcrow/atomic/tree/vendor-dir which move vendored code into ``./vendor`` as a start. Feel free to include it otherwise I'll make a different PR after this merges.

**Comment by giuseppe** [comment]
@vbatts does this change make any sense to you?  Do you know if Docker is going to use the gzip implementation from golang 1.8 and we can switch as well?

**Comment by vbatts** [comment]
oh interesting. I figured this day would come. It would be ripping a bandaid off to move to the newer gzip/flate. Is the go1.8 implementation consistent to itself, just not consistent to prior versions?

**Comment by giuseppe** [comment]
@vbatts yes, go1.8 is consistent to itself but it differs from golang1.6.

**Comment by vbatts** [comment]
it makes sense, but eventually will have conflicts with checksums by other systems if they haven't done the same. This is why many folks (e.g. me) have pushed to only compare checksums of the _uncompressed_ archive [where possible]. I think this is something docker have begun doing more of.

From a quick look at moby/moby master, they are not vendoring/forking this code and only importing from `compress/gzip`, so where they're relying on the checksum of a compressed archive, the hash will change when the daemon is compiled with newer golang.

**Comment by giuseppe** [comment]
Thanks.  If dockertar-sha256-helper was compiled with golang 1.8 then it is probably safe to assume the same version of golang was used for Moby or any other tool that calculates the checksum.  I am going to close this PR since there are no clear benefits and we start carrying a lot more code

**Pull request closed without merging**